app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/Utils.java

```diff
@@ -660,13 +665,13 @@ public static String getEnergy(String value) {
     }
 
     private static int convertKjToKcal(double kj) {
-        return kj != 0 ? Double.valueOf(kj / 4.1868d).intValue() : -1;
+        return kj != 0 ? (int) (kj / 4.1868d) : -1;
```

Comment:
I remember we had issues at some point with the API returning int and doubles at random. Not sure if fully fixed on all products @stephanegigandet @deniger

Reply:
this modification  is just done to avoid boxing double to Double then to int